### PR TITLE
fix: get score in util & get leaves from ipfs

### PR DESCRIPTION
**Pull request by ioanaroceanu2000**
- get leaves from IPFS when doing a redeem
- get score (for ranking) is now a util

### CODE REVIEW

File: src/utils/communitySbt/getIpfsLeaves.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    const snaphots : Array<{
+            owner: string
+            badgeType: number,
+            metadataURI: string
+        }> = data.data.snapshot;
+
+    let subgraphSnapshots: Array<LeafInfo> = [];
+
+    for (const entry of snaphots) {
```
Comment (ssbarbee) [inline src/utils/communitySbt/getIpfsLeaves.ts:20]: why not use map here? snaphots.map = ...

File: src/utils/communitySbt/getIpfsLeaves.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    const data = await axios.get(geLeavesIpfsUri(network, seasonId));
+
+    const snaphots : Array<{
+            owner: string
+            badgeType: number,
+            metadataURI: string
+        }> = data.data.snapshot;
+
+    let subgraphSnapshots: Array<LeafInfo> = [];
```
Comment (ssbarbee) [inline src/utils/communitySbt/getIpfsLeaves.ts:18]: this can be const

File: src/utils/communitySbt/getTopBadges.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            score = updateScore(swapArgs);
+          }
+        }
+  
+        if (score > 0 && !ignoredWalletIds[wallet.id.toLowerCase()]) {
+            scores[wallet.id] = score;
+        }
+      }
+  
+      if (data.data.wallets.length < 1000) {
```
Comment (ssbarbee) [inline src/utils/communitySbt/getTopBadges.ts:153]: what if data.data.wallets.length breaks due to data being null, data.data being null, data.data.wallets being null. I would use ? operator here

File: src/utils/communitySbt/seasonsConfig.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export const goerliSeasonLeavesCid: Record<number, string> = {
+    0: "QmYjnH1btsV5of8NHmjUUmK5va7kGBoVDTUwWy1HtLyq7d",
```
Comment (ssbarbee) [inline src/utils/communitySbt/seasonsConfig.ts:2]: goerliSeasonLeavesCid[0] is same as mainnetSeasonLeavesCid[0]... On purpose?

Comment (ioanaroceanu2000) [reply src/utils/communitySbt/seasonsConfig.ts:2]: on purpose -> there's a TODO below (after deployment on mainnet badges, replace CID)

File: src/utils/communitySbt/getIpfsLeaves.ts (hunk 1 above)
Comment (ioanaroceanu2000) [reply src/utils/communitySbt/getIpfsLeaves.ts]: replaced in c856fd63bdec36d55495e3009171c2af0029d365 :)

File: src/utils/communitySbt/getIpfsLeaves.ts (hunk 2 above)
Comment (ioanaroceanu2000) [reply src/utils/communitySbt/getIpfsLeaves.ts]: used const .. = ...map() c856fd63bdec36d55495e3009171c2af0029d365

File: src/utils/communitySbt/getTopBadges.ts (hunk 3 above)
Comment (ioanaroceanu2000) [reply src/utils/communitySbt/getTopBadges.ts]: true, I added a check on this: if null -> empty array (c856fd63bdec36d55495e3009171c2af0029d365)

**Review by ssbarbee — APPROVED** [review]

**Pull request merged**